### ISSUE TITLE
Some episodes/clips are missing audio from TV4play

### ISSUE THREAD

**Issue by pythonuser3856** [spaam/svtplay-dl#1494]
Versions used:

snap-shot 20220709_125010 / '--version' print-out: svtplay-dl 4.13-13-g7023e43
Python 3.8.10
Windows 8.1
FFmpeg 5.0.1

At testing the link https://www.tv4play.se/program/lotta-på-liseberg/del-6-säsong-2022/13746226
svtplay-dl attempts to get 1162 episodes and clips.

However, some of these don't include audio.
After filtering out the ones with no audio I have 1046 remaining episodes/clips.

Given that svtplay-dl have the media selection switches '--only-audio' and '--only-video',
my assumption has been that a downloaded media without using any of these always includes both audio and video.
Is that not the case ?
If so it would be good to have a switch to make svtplay-dl exclude download of media that don't include both.
If svtplay-dl cannot resolve absence of audio at the time of producing URLs, it would be good to have it as a function during post-processing.

**Comment by pythonuser3856** [comment]
I agree with that a good solution is to drop items that do not include both audio and video unless the user has requested something different.
svtplay-dl should at dropping an incomplete media inform the user about file name as well as the link attempted to cater for a use of '--all-episodes --include-clips'. The user need this info to be able to choose to get the media using '--only-audio' or '--only-video'.

I used the arguments '--all-episodes --include-clips --subtitle' and then used VLC to verify that there is only a video stream.
It's not that straight forward to provide you a link since they are invisible to the user at using the combination of '--all-episodes --include-clips'. Anyway, by using the alternative of '--all-episodes --subtitle --include-clips --get-only-url' I created a list and used that in a script to execute a sequence of individual invocations of svtplay-dl for each link.
Doing that I think the following examples from the url-list will render a downloaded media to be w/o audio:
273
274
275
278
279
280
281
282
283
284
285
286
287
288
290
291
292 no download, error msg: no video found
293
294
295
296
297
298
299
300 no download, error msg: no video found
301
302
303
I cut the execution here.

**Comment by spaam** [comment]
kollade på https://www.tv4play.se/program/lotta-p%C3%A5-liseberg/3425172 . den har inte ens video i deras spelare 🤔

**Comment by pythonuser3856** [comment]
An advantage from using svtplay-dl with the multi-download option is that you get a bunch of media at one point in time for later use. A lot of that would go away if you at the point in time would need to manually verify each media.
From the result I got among the 1162 attempts I was concerned with that svtplay-dl may not always get a complete media when it should have. For single media downloads it may be an option (?) to manually check the result after download to make sure that both video and audio are included but not in the general case.

As long as svtplay-dl is made to indicate that an incomplete media is erased and at the same time provides the user with information about file name and url, it's fine with me.
Since a use of multi-download cause the console window to scroll away and by that may hide such information from the user, it would be good to get it as a separate file.

**Comment by pythonuser3856** [comment]
It seems as if noone else of the svtplay-dl users care so much right now.

Why I think it would be good to also get hold of file names and URLs of incomplete media is that you will be able to manually determine if there is  pattern among the failing ones.
In the example I brought up it seems as if there are multiple videos of the same artist that fails. A speculation could be an issue with music licensing. But why make a silent video available ?

Anyway, erasing would be good together with a one-liner that incomplete files have been erased in the console window at the end of svtplay-dl execution.

**Comment by assholehoff** [comment]
>From the result I got among the 1162 attempts I was concerned with that svtplay-dl may not always get a complete media when it should have. For single media downloads it may be an option (?) to manually check the result after download to make sure that both video and audio are included but not in the general case.
A similar case drove me nuts about two years ago. I was going to watch a movie I had "recorded" from SVT Play a few weeks prior, and about four minutes in, the sound went dead. The movie was no longer available, so I couldn't retry the save. Sometime later I had this problem again, and made a tool to check the files after saving them: [*checktracks.py*](https://github.com/assholehoff/checktracks.py)
I identified maybe three cases of about 400.

I just experienced the bug again today, audio on https://www.svtplay.se/video/KM6pZd6/never-look-away disappears after about four minutes.

I can't get the sound, however many times I retry, but at least I no longer get the surprise weeks later.😅

**Comment by spaam** [comment]
@assholehoff tell me about what arguments you use  with svtplay so someone can reproduce the issue?  will be pretty hard to do it if i dont have it.

**Comment by assholehoff** [comment]
ffmpeg 4.4.4_1 via MacPorts on macOS 13.5.2. The problem goes back two years though, so at least macOS 11.

I use the following settings when recording film from SVT Play:
`svtplay-dl --exclude syntolkat --merge-subtitle --filename '{title}-{service}.{ext}' \
'url1' \
'url2' \
...`

I tried `--keep-original` and checked the audio stream in the separate file, but it is also only about four minutes long.
